staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -1049,6 +1063,30 @@ func (cfgCtlr *configController) maybeReapReadLocked(plName string, plState *pri
 	cfgCtlr.configQueue.Add(0)
 }
 
+// PriorityLevelMaxSeats returns the maximum seats that a request should occupy
+// for a given priority level. See priorityLevelMaxSeatsLocked for more details.
+func (cfgCtlr *configController) PriorityLevelMaxSeats(plName string) uint64 {
+	// This function should only be invoked when work estimator runs in startRequest,
+	// where cgfCtlr's lock is already acquired.
+	return cfgCtlr.priorityLevelMaxSeatsLocked(plName)
+}
+
+// priorityLevelMaxSeatsLocked returns the maximum seats that a request should occupy
+// for a given priority level. This is called from the work estimator to dynamically
+// determine max seats. The current implementation estimates max seats as
+// MIN(0.15 * nominalCL, nominalCL/handSize). This function can return arbitrarily
+// high value of max seats, but work estimator currently caps max seats at 10.
+// priorityLevelMaxSeatsLocked requires the cfgCtlr's lock to already be held,
+// which should always be the case since the work estimator is called in startRequest
+func (cfgCtlr *configController) priorityLevelMaxSeatsLocked(plName string) uint64 {
+	plState, ok := cfgCtlr.priorityLevelStates[plName]
+	if !ok {
+		return 0
```

Comment:
(1) Wow, zero seems pretty hostile.
(2) This `return` should never be reached, right?

Reply:
It _shouldn't_ but work estimator will use default max if this returns 0